FFmpegInterop/FFmpegInteropBuffer.h

```diff
@@ -0,0 +1,78 @@
+//*****************************************************************************
+//
+//	Copyright 2019 Microsoft Corporation
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	http ://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+//
+//*****************************************************************************
+
+#pragma once
+
+namespace winrt::FFmpegInterop::implementation
+{
+	class FFmpegInteropBuffer :
+		public implements<
+			FFmpegInteropBuffer,
+			Windows::Storage::Streams::IBuffer,
+			::Windows::Storage::Streams::IBufferByteAccess,
+			IMarshal>
+	{
+	public:
+		FFmpegInteropBuffer(_In_ AVBufferRef* bufRef);
+		FFmpegInteropBuffer(_In_ AVBufferRef_ptr bufRef) noexcept;
+		FFmpegInteropBuffer(_In_ AVPacket_ptr packet);
+		FFmpegInteropBuffer(_In_reads_(bufSize) AVBlob_ptr buf, _In_ uint32_t bufSize);
+		FFmpegInteropBuffer(_In_ std::vector<uint8_t>&& buf) noexcept;
+
+		// IBuffer
+		uint32_t Capacity() const noexcept;
+		uint32_t Length() const noexcept;
+		void Length(_In_ uint32_t length);
+
+		// IBufferByteAccess
+		STDMETHODIMP Buffer(_Outptr_ uint8_t** buf) noexcept;
+
+		// IMarshal

```

Comment:
Why does this implement IMarshal, is it used cross process? Can there be some info about this in the description?

Reply:
Yes, in app service scenarios the buffer is marshaled cross proc. I'll add a comment.

Reply:
Yes, in media source app service scenarios the buffer is marshaled cross proc. I'll add a comment about it.